src/datumaro/plugins/missing_annotation_detection.py

```diff
@@ -0,0 +1,149 @@
+            Typically, the distance metric is Intersection over Union (IoU), which is bounded between 0 and 1.
+        score_threshold: The minimum score required for an annotation to be considered
+            a candidate for missing annotations.
+        label_agnostic_matching: If set to false, annotations with different labels are not matched
+            to determine their spatial overlap. In the above example, ``label_agnostic_matching=False``
+            will produce ``model_predictions`` as is since ``Bbox(0, 0, 1, 1, label=1, attributes={"score": 1.0})``
+            has different label with ``Bbox(0, 0, 1, 1, label=0)``.
+    """
+
+    def __init__(
+        self,
+        extractor: IDataset,
+        launcher: Launcher,
+        batch_size: int = 1,
+        pairwise_dist: float = 0.75,
+        score_threshold: Optional[float] = None,
+        label_agnostic_matching: bool = True,
+    ):
+        super().__init__(extractor, launcher, batch_size, append_annotation=False)
+        self._score_threshold = score_threshold
+
+        class LabelAgnosticMatcherContext(IMatcherContext):
+            def get_any_label_name(self, ann: Annotation, label_id: int) -> str:
+                return ""
+
+        label_categories: LabelCategories = self.categories()[AnnotationType.label]
+
+        class LabelGnosticMatcherContext(IMatcherContext):
+            def get_any_label_name(self, ann: Annotation, label_id: int) -> str:
+                return label_categories[label_id]
+
+        self._support_matchers = {
+            AnnotationType.bbox: BboxMatcher(
+                pairwise_dist=pairwise_dist,
+                context=LabelAgnosticMatcherContext()
+                if label_agnostic_matching
+                else LabelGnosticMatcherContext(),
+                match_segments=match_segments_more_than_pair,
+            ),
+        }
+
+    def _yield_item(
+        self, batch: List[DatasetItem], inference: List[List[Annotation]]
+    ) -> Generator[DatasetItem, None, None]:
+        for item, annotations in zip(batch, inference):
+            self._check_annotations(annotations)
+            yield self.wrap_item(
+                item,
+                annotations=self._find_missing_anns(
+                    gt_anns=item.annotations,
+                    pseudo_anns=self._apply_score_threshold(annotations),
+                ),
+            )
+
+    def _apply_score_threshold(self, annotations: List[Annotation]) -> List[Annotation]:
+        if self._score_threshold is None:
+            return annotations
+
+        return [
+            ann for ann in annotations if ann.attributes.get("score", 1.0) > self._score_threshold
+        ]
+
+    def _find_missing_anns(
+        self, gt_anns: List[Annotation], pseudo_anns: List[Annotation]
+    ) -> List[Annotation]:
+        for ann in pseudo_anns:
+            ann.attributes["_pseudo_label_"] = True
```

Comment:
when it comes to me, both `_pseudo_label_` and `score` could contain mostly duplicated information. How about checking the existence of `psuedo_label_score` here? Of course, this will be used for `score` above.

Reply:
This flag was used to pick up the pseudo label annotations from the cluster later. https://github.com/openvinotoolkit/datumaro/pull/1049/commits/25a3817217c993a2f8cc46a90bf57a29639dfe72 changed it with introducing `ids_of_pseudo_labels` rather than injecting a new attribute to the annotation.